### ISSUE TITLE
A potential way to run full C/C++/Go/Node.JS programs and more on iOS?

### ISSUE THREAD

**Issue by zaptrem** [blinksh/blink#580]
Could Blink utilize a simple internal web server + WebKit's JIT capabilities through JavaScript to create an offline and persistent implementation of [Browsix](https://browsix.org/) ("Unix In Your Browser Tab") to unlock more powerful tools and potentially enable the use of arbitrary Unix development tools and programs on iOS? I stumbled across both this project and Browsix while looking into iPad development environments and thought the possibilities were interesting.

**Comment by zaptrem** [comment]
Another clunkier but potentially more finished/fully featured solution is JSLinux: https://bellard.org/jslinux/index.html

**Comment by yury** [comment]
Some progress on that
![431326AC-0B8E-4265-950E-BDBCEB8479AE](https://user-images.githubusercontent.com/5250/54132452-501bf900-4425-11e9-9973-8854af91ec2d.png)
![606EB270-FF4B-4E06-BC89-99E23BB69658](https://user-images.githubusercontent.com/5250/54132454-50b48f80-4425-11e9-9d23-2e512d6ce261.png)

**Comment by diogoviannaaraujo** [comment]
hi @yury is there a public development branch of this node running version somewhere?

**Comment by yury** [comment]
Nope, I stopped experiments for this for now. Turns out it very hard to make it thread safe (to run multiple node engines)

**Comment by zaptrem** [comment]
@yury Wouldn’t even a single threaded version of this be significantly faster than what is available now?
